src/changes.go

```diff
@@ -262,6 +262,7 @@ type changeSliceArg struct {
 	localParent, remoteParent string
 	dirList                   []*dirList
 	changeListPtr             *[]*Change
+	mtx                       *sync.Mutex
```

Comment:
changing to mu is fine. I'll send an update for that.

The non-pointer I think will be an issue. Each go routine needs to be using the same mutex to effectively communicate. So if I pass the mutex by value, I don't think they will be. I did a quick test and -race was still showing the race condition.

The other option would be to pass a channel and have the workers return values via the channel and add them to the list using a single routine.